docs/tutorials/01-intro.md

```diff
@@ -0,0 +1,209 @@
+
+In brief the `containers/image` project provides mechanisms to copy, push, pull, inspect and sign container images. The `containers/storage` project provides mechanisms for storing filesystem layers, container images, and containers. `buildah` is a CLI that takes advantage of these underlying projects and therefore allows you to build, move, and manage container images and containers.  
+
+First step is to install buildah:
+
+    dnf -y install buildah
+
+After installing buildah we can see there are no images installed. The `buildah images` command will list all the images:
+
+    buildah images
+
+We can also see that there are also no containers by running:
+
+    buildah containers
+  
+When you build a working container from an existing image, buildah defaults to appending '-working-container' to the image's name to construct a name for the container. The buildah CLI conveniently returns the name of the new container. You can take advantage of this by assigning the returned value to a shell varible using standard shell assignment :
+
+    container=$(buildah from fedora)
+
+Of course you can always just run `buildah from fedora` and then list the containers to see the container name. As mentioned previously it will be fedora-working-container in this case.
+
+See the name:  
+
+    echo $container
+
+What can we do with this new container? Let's try running bash:
+
+    buildah run $container bash
+    
+Notice we get a new shell prompt because we are running a bash shell inside of the container. It should be noted that `buildah run` is not intended for running production containers. It is for helping debug during the build process. 
+
+Let's try running something else:
+
+    buildah run $container java
+
+Oops. Java is not installed. A message containing something like the following was returned.
+
+    container_linux.go:274: starting container process caused "exec: \"java\": executable file not found in $PATH"
+    
+Lets try installing it using:
+    
+    buildah run $container -- dnf install java
+
+The `--` syntax basically says: no more `buildah run` command options after this point. It is required if the command we specify includes command line options which are not meant for buildah. 
+
+Now running `buildah run $container java` will show that Java has been installed. It will return the `Usage`:
+
+## Building a container from scratch
+
+One of the advantages of using `buildah` to build OCI compliant container images is that you can easily build a container image from scratch and therefore exclude unnecessary packages from your image. E.g. most final container images for production probably don't need a package manager like `dnf`. 
+
+Let's build a container from scratch. The special "image" name scratch tells buildah to create an empty container:
```

Comment:
IDK if you want to get into nits or not, but the container is not completely empty.  It has a practically negligible amount of meta data stored in it so it will have enough smarts to start up.  Practically immeasurable.

Reply:
I think this is useful. I've made an appropriate change.